internal/opa/validator.go

```diff
@@ -0,0 +1,296 @@
+		return nil, fmt.Errorf("failed to load policies: %w", err)
+	}
+
+	zap.S().Named("opa").Infof("OPA validator initialized with policies from: %s", policiesDir)
+	return validator, nil
+}
+
+func (v *Validator) loadPolicies(policiesDir string) error {
+	policies := make(map[string]string)
+
+	entries, err := os.ReadDir(policiesDir)
+	if err != nil {
+		return fmt.Errorf("failed to read policies directory: %w", err)
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() || !strings.HasSuffix(entry.Name(), ".rego") ||
+			strings.HasSuffix(entry.Name(), "_test.rego") {
+			continue // Skip test files
+		}
+
+		path := filepath.Join(policiesDir, entry.Name())
+		content, err := os.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("failed to read policy file %s: %w", path, err)
+		}
+
+		policies[entry.Name()] = string(content)
+		zap.S().Named("opa").Debugf("Loaded policy: %s", entry.Name())
+	}
+
+	if len(policies) == 0 {
+		return fmt.Errorf("no .rego policy files found in directory: %s", policiesDir)
+	}
+
+	compiler := ast.NewCompiler()
+	modules := make(map[string]*ast.Module)
+
+	for filename, content := range policies {
+		// Parse with v0 for compatibility with existing Forklift policies
+		module, err := ast.ParseModuleWithOpts(filename, content, ast.ParserOptions{
+			RegoVersion: ast.RegoV0,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to parse policy %s: %w", filename, err)
+		}
+		modules[filename] = module
+	}
+
+	compiler.Compile(modules)
+	if compiler.Failed() {
+		return fmt.Errorf("policy compilation failed: %v", compiler.Errors)
+	}
+
+	// Use v1 runtime for future-proofing and better performance
+	r := rego.New(
+		rego.Query("data.io.konveyor.forklift.vmware.concerns"),
+		rego.Compiler(compiler),
+		rego.SetRegoVersion(ast.RegoV1),
+	)
+
+	preparedQuery, err := r.PrepareForEval(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed to prepare rego query: %w", err)
+	}
+
+	v.preparedQuery = preparedQuery
+	zap.S().Named("opa").Infof("Successfully compiled %d policy files", len(policies))
+	return nil
+}
+
+func (v *Validator) ValidateConcerns(ctx context.Context, input interface{}) ([]interface{}, error) {
+	resultSet, err := v.preparedQuery.Eval(ctx, rego.EvalInput(input))
+	if err != nil {
+		return nil, fmt.Errorf("policy evaluation failed: %w", err)
+	}
+
+	if len(resultSet) == 0 || len(resultSet[0].Expressions) == 0 {
+		zap.S().Named("opa").Debug("No policy results returned")
+		return []interface{}{}, nil
+	}
+
+	result, ok := resultSet[0].Expressions[0].Value.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("unexpected result type from policy evaluation")
+	}
+
+	return result, nil
+}
+
+// DiscoverPoliciesDirectory finds the policies directory using standard locations
+// and auto-downloads policies if not found locally
+func DiscoverPoliciesDirectory() string {
+	logger := zap.S().Named("opa")
+	candidates := []string{
+		"./policies",    // Local development
```

Comment:
Could we have this configurable via env var?  Where default is `/app/policies`? and on dev-env you are suppose to set the var?